src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -196,7 +211,7 @@ internal class ModelCheckingStrategy(
      */
     private inner class SwitchChoosingNode : InterleavingTreeNode() {
         override fun nextInterleaving(interleavingBuilder: InterleavingBuilder): Interleaving {
-            val isLeaf = maxNumberOfSwitches == interleavingBuilder.numberOfSwitches
+            val isLeaf = maxNumberOfSwitches == interleavingBuilder.numberOfCrashes + interleavingBuilder.numberOfSwitches
```

Comment:
Probably, instead of `numberOfCrashes` or `numberOfSwitches` we should be just `numberOfEvents`